src/cargo/util/toml/targets.rs

```diff
@@ -184,6 +185,12 @@ pub fn normalize_lib(
             }
         }
 
+        if let Some(PathValue(path)) = lib.path.as_ref() {
+            let path = package_root.join(path);
+            let path = paths::normalize_path(&path);
+            lib.path = Some(PathValue(path.into()));
+        }
```

Comment:
If we're doing the `join`s here, should we remove the `join`s in the `to_` functions?